tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -1433,6 +1433,52 @@ template <typename T> T wavePrefixProduct(T A, UINT) {
   return static_cast<T>(A * A);
 }
 
+template <typename T> struct Op<OpType::WaveMatch, T, 1> : StrictValidation {};
+
+template <typename T> struct ExpectedBuilder<OpType::WaveMatch, T> {
+  static std::vector<UINT> buildExpected(Op<OpType::WaveMatch, T, 1> &,
+                                         const InputSets<T> &Inputs,
+                                         UINT WaveSize) {
+    DXASSERT_NOMSG(Inputs.size() == 1);
+
+    std::vector<UINT> Expected;
+    const size_t VectorSize = Inputs[0].size();
+    Expected.assign(VectorSize, 0);
+
+    UINT wordShift = WaveSize / 32;
+    UINT bitShift = WaveSize % 32;
+    if (bitShift == 0) {
+      bitShift = 32;
+      wordShift--;
+    }
+    uint64_t result[4] = {((1ULL << bitShift) - 1) & ~1ULL, 0, 0, 0};
```

Comment:
Did you mean for this to be uint32_t?

Reply:
It's really confusing this not being uint32_t.